src/ci/scripts/free-disk-space.sh

```diff
@@ -0,0 +1,218 @@
+
+BEFORE=$(getAvailableSpace)
+
+# https://github.community/t/bigger-github-hosted-runners-disk-space/17267/11
+sudo rm -rf /usr/share/dotnet || true
+
+AFTER=$(getAvailableSpace)
+SAVED=$((AFTER-BEFORE))
+printSavedSpace $SAVED ".NET runtime"
+
+# Remove Haskell runtime
+BEFORE=$(getAvailableSpace)
+
+sudo rm -rf /opt/ghc || true
+sudo rm -rf /usr/local/.ghcup || true
+
+AFTER=$(getAvailableSpace)
+SAVED=$((AFTER-BEFORE))
+printSavedSpace $SAVED "Haskell runtime"
+
+# Remove large packages
+# REF: https://github.com/apache/flink/blob/master/tools/azure-pipelines/free_disk_space.sh
+
+BEFORE=$(getAvailableSpace)
+
+sudo apt-get remove -y '^aspnetcore-.*' || echo "::warning::The command [sudo apt-get remove -y '^aspnetcore-.*'] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y '^dotnet-.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y '^dotnet-.*' --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y '^llvm-.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y '^llvm-.*' --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y 'php.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y 'php.*' --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y '^mongodb-.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y '^mongodb-.*' --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y '^mysql-.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y '^mysql-.*' --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y azure-cli google-chrome-stable firefox powershell mono-devel libgl1-mesa-dri --fix-missing || echo "::warning::The command [sudo apt-get remove -y azure-cli google-chrome-stable firefox powershell mono-devel libgl1-mesa-dri --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y google-cloud-sdk --fix-missing || echo "::debug::The command [sudo apt-get remove -y google-cloud-sdk --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y google-cloud-cli --fix-missing || echo "::debug::The command [sudo apt-get remove -y google-cloud-cli --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y microsoft-edge-stable --fix-missing || echo "::debug::The command [sudo apt-get remove -y microsoft-edge-stable --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get remove -y snapd --fix-missing || echo "::debug::The command [sudo apt-get remove -y snapd --fix-missing] failed to complete successfully. Proceeding..."
+sudo apt-get autoremove -y || echo "::warning::The command [sudo apt-get autoremove -y] failed to complete successfully. Proceeding..."
+sudo apt-get clean || echo "::warning::The command [sudo apt-get clean] failed to complete successfully. Proceeding..."
```

Comment:
```suggestion
PACKAGES_TO_REMOVE=(
    '^aspnetcore-.*'
    '^dotnet-.*'
    '^llvm-.*'
    'php.*'
    '^mongodb-.*'
    '^mysql-.*'
    'azure-cli'
    'google-chrome-stable'
    'firefox'
    'powershell'
    'mono-devel'
    'libgl1-mesa-dri'
    'google-cloud-sdk'
    'google-cloud-cli'
    'microsoft-edge-stable'
    'snapd'
)

for pkg in "${PACKAGES_TO_REMOVE[@]}"; do
    sudo apt-get remove -y "$pkg" --fix-missing || echo "::warning::Failed to remove $pkg"
done

sudo apt-get autoremove -y || echo "::warning::The command [sudo apt-get autoremove -y] failed"
sudo apt-get clean || echo "::warning::The command [sudo apt-get clean] failed failed"
```

We can simplify the code like this. What do you think?
The difference is that `'^aspnetcore-.*'` didn't have `--fix-missing` but I'm not sure why, so we can try to add it if you agree 👍